File: about-java/app/src/main/java/com/example/aboutjava/MainActivity.java
package com.example.aboutjava;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.media.midi.MidiDeviceService;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    Button inheritance;
    Button abstraction;
    Button polymorphism;
    Button encapsulation;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        inheritance = (Button) findViewById(R.id.inheritance);
        inheritance.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                Intent i1 = new Intent(MainActivity.this, InheritanceJava.class);
                startActivity(i1);
            }
        });

        abstraction = (Button) findViewById(R.id.abstraction);
        abstraction.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                Intent i2 = new Intent(MainActivity.this, AbstractionJava.class);
                startActivity(i2);
            }
        });

        polymorphism = (Button) findViewById(R.id.polymorhism);
        polymorphism.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                Intent i3 = new Intent(MainActivity.this, PolymorsphismJava.class);
                startActivity(i3);
            }
        });

        encapsulation = (Button) findViewById(R.id.encapsulation);
        encapsulation.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                Intent i4 = new Intent(MainActivity.this, EncapsulationJava.class);
                startActivity(i4);
            }
        });
    }
}

File: database-concept/app/src/main/java/com/example/database/MainActivity.kt
package com.example.database

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.webkit.WebViewDatabase
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import com.google.android.material.textfield.TextInputEditText
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase

class MainActivity : AppCompatActivity() {

    lateinit var database: DatabaseReference
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val signInButton = findViewById<Button>(R.id.btnSignUp)
        val etName = findViewById<TextInputEditText>(R.id.etName)
        val etEmail = findViewById<TextInputEditText>(R.id.etEmail)
        val etPassword = findViewById<TextInputEditText>(R.id.etPassword)
        val etUniqueId = findViewById<TextInputEditText>(R.id.etUniqueId)

        signInButton.setOnClickListener {

            val name = etName.text.toString()
            val mail = etEmail.text.toString()
            val uniqueId = etUniqueId.text.toString()
            val password = etPassword.text.toString()


            val users = Users(name, mail, password, uniqueId)
            database = FirebaseDatabase.getInstance().getReference("Users")

            database.child(uniqueId).setValue(users).addOnSuccessListener {

                Toast.makeText(this, "User Registered", Toast.LENGTH_SHORT).show()
            }.addOnSuccessListener {
                Toast.makeText(this, "Failed", Toast.LENGTH_SHORT).show()

            }
        }

    }
}